db/hints/sync_point.cc

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2021-present ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <exception>
+#include <unordered_set>
+
+#include <seastar/core/simple-stream.hh>
+
+#include "db/hints/sync_point.hh"
+#include "gms/inet_address_serializer.hh"
+#include "idl/uuid.dist.hh"
+#include "idl/uuid.dist.impl.hh"
+#include "idl/replay_position.dist.hh"
+#include "idl/replay_position.dist.impl.hh"
+#include "idl/hinted_handoff.idl.hh"
+#include "idl/hinted_handoff.dist.hh"
+#include "idl/hinted_handoff.dist.impl.hh"
+#include "serializer.hh"
+#include "serializer_impl.hh"
+#include "bytes.hh"
+
+namespace db {
+namespace hints {
+
+// TODO: Encode host ID so that two sync points won't get mixed up
```

Comment:
It looks like host id is already included in the sync point (although not in per_manager_sync_point), is this comment up to date?

Also, just curious - do you include the host id just to avoid a situation when a client gets the sync point from one node, and then mistakenly tries to wait on it on another node?

Reply:
> It looks like host id is already included in the sync point (although not in per_manager_sync_point), is this comment up to date?

No, it's not up to date. I forgot to remove it after I implemented the host ID check. I'll remove the comment.

> Also, just curious - do you include the host id just to avoid a situation when a client gets the sync point from one node, and then mistakenly tries to wait on it on another node?

Yes. It doesn't make sense to use a sync point created on one node to wait on another node. This check is verified in the dtest.